pkg/storage/stores/shipper/indexgateway/gateway.go

```diff
@@ -218,10 +219,16 @@ func (g *Gateway) LabelValuesForMetricName(ctx context.Context, req *logproto.La
 	// An empty matchers string cannot be parsed,
 	// therefore we check the string representation of the the matchers.
 	if req.Matchers != syntax.EmptyMatchers {
-		matchers, err = syntax.ParseMatchers(req.Matchers)
+		expr, err := syntax.ParseExprWithoutValidation(req.Matchers)
 		if err != nil {
 			return nil, err
 		}
+
+		matcherExpr, ok := expr.(*syntax.MatchersExpr)
+		if !ok {
+			return nil, errors.New("invalid label matchers found")
```

Comment:
```suggestion
			return nil, fmt.Errorf("invalid label matchers found of type %T", expr)
```

I'm hoping we have another layer of indirection above this that gives a more human-readable response though